eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -86,53 +86,69 @@ stages:
                           arguments: >-
-                            -GitHubPat $(azuresdk-github-pat)
                             -GitHubRepo "$(Build.Repository.Name)"
                             -PackageSpecPath $(Pipeline.Workspace)/packages/${{artifact.Name}}/package-info.json
                             -SourceDirectory $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg
                             -Verbose
                           pwsh: true
                         displayName: Initialize vcpkg release
 
+                      # There are potential race conditions if this script runs
+                      # in parallel against the same branch name. Release only 
+                      # one package at a time.
                       - pwsh: |
+                          Write-Host "git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg"
                           git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg
                           if ($LASTEXITCODE -ne 0) {
                             Write-Error "Unable to check out vcpkg repo"
                             exit $LASTEXITCODE
                           }
+                          
+                          Write-Host "cd $(Pipeline.Workspace)/vcpkg"
                           cd $(Pipeline.Workspace)/vcpkg
 
-                          # There is no need to check out a branch, the
-                          # create-pull-request.yml template creates the branch
-
                           # Clean out the folder so that template files removed
                           # are not inadvertently re-added
-                          if (Test-Path "ports/${{ artifact.Name }}") {
-                            Remove-Item -v -r "ports/${{ artifact.Name }}"
+                          if (Test-Path "ports/${{ artifact.VcpkgPortName }}") {
+                            Remove-Item -v -r "ports/${{ artifact.VcpkgPortName }}"
                           }
 
-                          New-Item -Type Directory ports/${{ artifact.Name }}
-                          Copy-Item -Verbose $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg/port/* ports/${{ artifact.Name }}
+                          New-Item -Type Directory ports/${{ artifact.VcpkgPortName }}
+                          Copy-Item -Verbose $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg/port/* ports/${{ artifact.VcpkgPortName }}
 
                           # Show artifacts copied into ports folder for PR
-                          Get-ChildItem -Recurse ports/${{ artifact.Name }}
+                          Get-ChildItem -Recurse ports/${{ artifact.VcpkgPortName }}
+                          
+                          Write-Host "git status"
+                          git status
                         displayName: Copy updated files into vcpkg fork for PR
                         condition: ne(variables['SkipVcpkgUpdate'], 'true')
```

Comment:
Is this condition still interesting? If so we should probably also need to include the other steps are part of that condition and we should also probably add `succeeded()` to it.

Reply:
Yes. There may be situations where we don't have a vcpkg port for a package. In that case we don't want to publish to vcpkg. Adding `succeeded()` is always the right thing to do here, as well. 

If we do need to skip that happens in https://github.com/Azure/azure-sdk-for-cpp/blob/01740c92555dc4457045cc4ae06e32894d07ec53/eng/scripts/Initialize-VcpkgRelease.ps1#L35 

Another approach might be to use the presence of `parameters.VcpkgPortName` to decide whether to create a vcpkg publish step.

Reply:
I'm mostly wanted to be sure that other steps like the create-pull request etc don't end up failing in the case we want to skip.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1356